app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,130 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val _popularSelectedCategory = MutableStateFlow(MovieCategory.POPULAR_STREAMING)
+    private val _nowPlayingSelectedCategory = MutableStateFlow(MovieCategory.PLAYING_MOVIES)
+    private val _upcomingSelectedCategory = MutableStateFlow(MovieCategory.UPCOMING_TODAY)
+
+    private val _popularCategoryViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val popularCategoryViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularCategoryViewState.asStateFlow()
```

Comment:
again, same comment as for other view models, you can remove redundancy like this:

```kotlin
private val popularMoviesCategorySelected = MutableStateFlow(MovieCategory.POPULAR_STREAMING)

val popularCategoryViewState: StateFlow<HomeMovieCategoryViewState> =
    popularMoviesCategorySelected
        .flatMapLatest { selectedMovieCategory ->
            movieRepository.popularMovies(selectedMovieCategory)
                .map { popularMovies ->
                    homeScreenMapper.toHomeMovieCategoryViewState(
                        movieCategories = listOf(
    MovieCategory.POPULAR_STREAMING,
    MovieCategory.POPULAR_ONTV,
    MovieCategory.POPULAR_RENT,
    MovieCategory.POPULAR_THEATERS
),
                        selectedMovieCategory = selectedMovieCategory,
                        movies = popularMovies,
                    )
                }
        }
        .stateIn(
            scope = viewModelScope,
            started = SharingStarted.WhileSubscribed(5_000),
            initialValue = homeScreenMapper.toHomeMovieCategoryViewState(
                movieCategories = popularMovieCategories,
                selectedMovieCategory = popularMoviesCategorySelected.value,
                movies = emptyList(),
            )
        )
```

Reply:
also, do this for other view states